storage/tests/SMAIndexSubBlock_unittest.cpp

```diff
@@ -0,0 +1,998 @@
+/**
+ *   Copyright 2016 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <limits>
+#include <memory>
+#include <set>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "catalog/CatalogAttribute.hpp"
+#include "catalog/CatalogRelation.hpp"
+#include "catalog/CatalogTypedefs.hpp"
+#include "expressions/predicate/ComparisonPredicate.hpp"
+#include "expressions/predicate/PredicateCost.hpp"
+#include "expressions/scalar/ScalarAttribute.hpp"
+#include "expressions/scalar/ScalarLiteral.hpp"
+#include "storage/CompressedColumnStoreTupleStorageSubBlock.hpp"
+#include "storage/SMAIndexSubBlock.hpp"
+#include "storage/StorageBlockInfo.hpp"
+#include "storage/StorageBlockLayout.pb.h"
+#include "storage/TupleIdSequence.hpp"
+#include "storage/tests/MockTupleStorageSubBlock.hpp"
+#include "types/CharType.hpp"
+#include "types/DoubleType.hpp"
+#include "types/FloatType.hpp"
+#include "types/LongType.hpp"
+#include "types/TypeFactory.hpp"
+#include "types/TypeID.hpp"
+#include "types/TypedValue.hpp"
+#include "types/VarCharType.hpp"
+#include "types/containers/Tuple.hpp"
+#include "types/operations/comparisons/Comparison.hpp"
+#include "types/operations/comparisons/ComparisonFactory.hpp"
+#include "types/operations/comparisons/ComparisonID.hpp"
+#include "utility/ScopedBuffer.hpp"
+
+#include "gtest/gtest.h"
+
+using std::binary_search;
+using std::int64_t;
+using std::numeric_limits;
+using std::ostringstream;
+using std::set;
+using std::size_t;
+using std::string;
+using std::uint8_t;
+using std::uint16_t;
+using std::uint32_t;
+using std::vector;
+
+namespace quickstep {
+
+typedef sma_internal::EntryReference EntryReference;
+typedef sma_internal::SMAEntry SMAEntry;
+typedef sma_internal::SMAPredicate SMAPredicate;
+typedef sma_internal::Selectivity Selectivity;
+
+namespace sma_test {
+  // Handy comparison functions which are used in the unit tests.
+
+  bool longs_equal(TypedValue left, TypedValue right) {
```

Comment:
We generally use ``camelCase`` instead of ``underscore_name`` to name functions.